pkg/clients/gitlab/git.go

```diff
@@ -50,14 +50,19 @@ func (gc *GitlabClient) ExistsBranch(projectID, branchName string) (bool, error)
 
 // DeleteBranch deletes a branch by its name and project ID
 func (gc *GitlabClient) DeleteBranch(projectID, branchName string) error {
-
-	_, err := gc.client.Branches.DeleteBranch(projectID, branchName)
+	resp, err := gc.client.Branches.DeleteBranch(projectID, branchName)
 	if err != nil {
+		// Check if the error is due to the branch not existing
+		if resp != nil && resp.StatusCode == http.StatusNotFound {
+			// Branch does not exist, log a message and return nil
+			fmt.Printf("Branch %s in projectID %s does not exist\n", branchName, projectID)
+			return nil
```

Comment:
same here

Reply:
same answer as in the former ,  @dheerajodha wdyt?